components/metricsView/MetricsView.component.tsx

```diff
@@ -1,65 +1,65 @@
 import { GetInfoNode } from "../../model/GetInfoNode";
 import Offline from "../genericView/Offline.component";
-import useSWR from "swr";
-import { fetcher } from "../../utils/AppUtils";
-import Loading from "../genericView/Loading.component";
 import React from "react";
 import Grid from "@mui/material/Grid";
 import SummaryChannels from "./cards/SummaryChannels.components";
 import UpTimeNode from "./cards/UpTimeNode.component";
 import ChannelsPayments from "./cards/ChannelsPayments.component";
 import theme from "../../theme/DarkTheme";
 import { Box } from "@mui/material";
+import {
+  ApolloClient,
+  InMemoryCache,
+  ApolloProvider,
+  gql,
+} from "@apollo/client";
 
 type MetricsViewProps = {
   nodeInfo: GetInfoNode | null;
   show: (visible: boolean, message: string) => void;
 };
 
-export default function MetricsView({
-  nodeInfo,
-  show,
-}: MetricsViewProps): JSX.Element {
-  const { data, error } = useSWR("/api/metrics", fetcher);
-  if (!nodeInfo) return <Offline />;
-  if (error) {
-    //TODO adding an error view
-    show(true, `Error: ${error.toString()}`);
-    return <></>;
-  }
-  if (!data) return <Loading />;
-  let metricsOne: any = data.metrics["metric_one"];
-  console.debug(`Metric one with the follow payload ${metricsOne.toString()}`);
+const client = new ApolloClient({
+  uri: "https://api.lnmetrics.info/query",
+  cache: new InMemoryCache(),
+});
```

Comment:
this is wrong, we are an API provider that is able to do that, you need to create the client inside the API.

You also note that there is already a getMetrics method.

I think you did not full catch why API provider exist?

Reply:
Changed the implementation